api-flask/app/ews.py

```diff
@@ -0,0 +1,144 @@
+"""Collect and parse Cambodia EWS-1294."""
+import itertools
+from datetime import datetime, timedelta, timezone
+
+from dateutil.parser import parse as dtparser
+
+from flask import request
+
+import numpy
+
+import requests
+
+from werkzeug.exceptions import BadRequest
+
+
+def parse_ews_params():
+    """Transform params used for ews request."""
+    only_dates = True if request.args.get('onlyDates') else False
+
+    today = datetime.now().replace(tzinfo=timezone.utc)
+    begin_datetime_str = request.args.get('beginDateTime')
+    if begin_datetime_str is not None:
+        begin_datetime = dtparser(begin_datetime_str)
+    else:
+        # yesterday
+        begin_datetime = today
+    begin_datetime = begin_datetime.replace(tzinfo=timezone.utc)
+
+    end_datetime_str = request.args.get('endDateTime')
+    if end_datetime_str is not None:
+        end_datetime = dtparser(end_datetime_str)
+    else:
+        # today
+        end_datetime = today
+    end_datetime = end_datetime.replace(tzinfo=timezone.utc)
+
+    # strptime function includes hours, minutes, and seconds as 00 by default.
+    # This check is done in case the begin and end datetime values are the same.
+    if end_datetime == begin_datetime:
+        end_datetime = end_datetime + timedelta(days=1)
+
+    if begin_datetime > end_datetime:
+        raise BadRequest('beginDateTime value must be lower than endDateTime')
+
+    if begin_datetime > today:
+        raise BadRequest('beginDateTime value must be less or equal to today')
+
+    return only_dates, begin_datetime, end_datetime
+
+
+def get_ews_responses(only_dates, begin_datetime, end_datetime):
+    """Get all data using ews_1294 api endpoints."""
+    # NOTE: Since ews-1294 api have performance issues, we decided to take shortcut
```

Comment:
I'd re-write this. We're not querying a single day purely because of performance issues. The PRISM front-end will initial display a single day's worth of data for all points. Only after 'drilling down' to the sensor level would we need a longer date range